nameko/messaging.py

```diff
@@ -310,13 +309,23 @@ def unregister_provider(self, provider):
 
         super(QueueConsumer, self).unregister_provider(provider)
 
-    @retry(for_exceptions=RecoverableConnectionError, max_attempts=3)
     def ack_message(self, message):
-        message.ack()
+        # only attempt to ack if the message connection is alive;
+        # otherwise the message will already have been reclaimed by the broker
```

Comment:
does the broker reclaim immediately? (i.e. no grace period?)

Reply:
As soon as it detects the connection is closed, yes. I don't think a grace period would make any difference.

Reply:
didn't mean for us to leave a grace period, but if the broker did (if so, we could try and just detect failures) but this is probably fine